lib/Doctrine/ORM/Persisters/SqlExpressionVisitor.php

```diff
@@ -80,6 +81,10 @@ public function walkCompositeExpression(CompositeExpression $expr)
             case CompositeExpression::TYPE_OR:
                 return '(' . implode(' OR ', $expressionList) . ')';
 
+            // Multiversion support for `doctrine/collections` before and after v2.1.0
+            case defined(CompositeExpression::class . '::TYPE_NOT') ? CompositeExpression::TYPE_NOT : 'NOT':
+                return 'NOT (' . $expressionList[0] . ')';
```

Comment:
I don't fully understand this change. Why are we falling back to `'NOT'` here? If the constant is not defined, doesn't this mean we cannot support that expression type?

Reply:
You're right. But because I'm in a `switch` structure, I couldn't define a `case` that should only be available if the constant exists, unless modifying it a bit more. Would you suggest I change its structure to something like this ?
```
$type = $expr->getType();
switch (true) {
    case ($type === CompositeExpression::TYPE_AND):
        return 'NOT (' . $expressionList[0] . ')';
    case ($type === CompositeExpression::TYPE_OR):
        return '(' . implode(' OR ', $expressionList) . ')';
    case (defined(CompositeExpression::class . '::TYPE_NOT') && $type === CompositeExpression::TYPE_NOT):
        return return 'NOT (' . $expressionList[0] . ')';
    default:
        throw new RuntimeException('Unknown composite ' . $type);
}
```

Reply:
I'd keep the `switch` as it was before and handle your new case inside the default block.

Reply:
I moved it to the default case. Tell me if there's anything else I can do.